Tasks/Common/AzureRmDeploy-common/azure-arm-rest/azure-arm-app-service.ts

```diff
@@ -204,16 +206,28 @@ export class AzureAppService {
         }
     }
 
-    public async patchApplicationSettings(addProperties: any, deleteProperties?: any): Promise<boolean> {
+    public async patchApplicationSettings(addProperties: any, deleteProperties?: any, formatJSON?: boolean): Promise<boolean> {
         var applicationSettings = await this.getApplicationSettings();
         var isNewValueUpdated: boolean = false;
-        for(var key in addProperties) {
-            if(applicationSettings.properties[key] != addProperties[key]) {
-                tl.debug(`Value of ${key} has been changed to ${addProperties[key]}`);
-                isNewValueUpdated = true;
+        if(formatJSON) {
+            for(var key in addProperties) {
+                if(JSON.stringify(applicationSettings.properties[key]) != JSON.stringify(addProperties[key])) {
+                    tl.debug(`Value of ${key} has been changed to ${JSON.stringify(addProperties[key])}`);
+                    isNewValueUpdated = true;
+                }
+
+                applicationSettings.properties[key] = addProperties[key];
```

Comment:
you should use JSON.stringify here as well?

Reply:
JSON.stringify was only used before to compare/display strings. The settings here are being passed as request body for API call.